src/generic/integral.h

```diff
@@ -1055,9 +1055,35 @@ namespace oomph
     /// Array to hold the weights and knots (defined in cc file)
     static const double Knot[52][2], Weight[52];
 
+#ifdef PARANOID
+    /// A flag to track whether we have warned the user about the exterior knots
+    /// and negative weights in this scheme.
+    static bool User_has_been_warned;
```

Comment:
Maybe I'm blind but where is this initialised to false?

Reply:
Woops! I've just forgotten to commit that change. It is in the .cc file (I'll commit it now).